test/lib/juttle-subprocess.spec.js

```diff
@@ -0,0 +1,152 @@
+var _ = require('underscore');
+var expect = require('chai').expect;
+var retry = require('bluebird-retry');
+
+function stringifyPoints(points) {
+    return _.map(points, function(point) {
+        point.time = point.time.valueOf();
+        return point;
+    });
+}
+
+describe('juttle-subprocess', function() {
+
+    var oldArgv;
+    var oldExit;
+    var exitStatus;
+    var sentData;
+
+    var findMessage = function(fields) {
+        return _.find(sentData, function(item) {
+            return _.reduce(fields, function(memo, value, key) {
+                return (item[key] === value) && memo;
+            }, true);
+        });
+    };
+
+    var waitForMessage = function(fields) {
+        return retry(function() {
+            if (!findMessage(fields)) {
+                throw new Error('Message not found with fields ' +
+                                JSON.stringify(fields) + ' in ' +
+                                JSON.stringify(sentData));
+            }
+        }, { max_tries: 10, interval: 100 });
+    };
+
+    beforeEach(function() {
+        oldArgv = process.argv;
+        process.argv = ['node', 'juttle-subproces.js', ''];
```

Comment:
Ah yes, this little trick is to make the subprocess happy when grabbing the `argv[2]` so the typo "doesn't matter" but I'll correct it.